src/components/Tasks/Tasks.tsx

```diff
@@ -70,19 +69,40 @@ const Tasks = () => {
       ? filterTasks(isUrgent)(selectedCategory)
       : tasks;
 
-  const displayTasks = filteredTasks.map((item) => {
-    return (
-      <Task
-        key={item.task}
-        {...item}
-        markCompleted={markCompleted}
-        deleteTask={deleteTask}
-      />
-    );
-  });
+   //useMemo
+  const displayTasks = useMemo(()=>{
+    const deleteTask = (task: string) => {
+      const filteredTasks = tasks.filter((element) => element.task !== task);
+      setTasks(filteredTasks);
+      localStorage.setItem("tasks", JSON.stringify(filteredTasks));
+    };
+
+    return filteredTasks.map((item) => {
+      return (
+        <Task
+          key={item.task}
+          {...item}
+          markCompleted={markCompleted}
+          deleteTask={deleteTask}
+        />
+      );
+    });
+  },[filteredTasks, tasks])
+
+  //useMemo
+  const displayContext = useMemo(()=>{
+    return contextData?.map((buddy: Buddy, index) => {
+      return (
+        <div className="buddy-smaller" key={index}>
+          <img src={buddy.url} alt="buddy-image" />
+        </div>
+      );
+    });
+  },[contextData])
 
   return (
     <div className="tasks">
+      <div className="buddies-smaller">{contextData && displayContext}</div>
```

Comment:
Do you see a problem if simplify it like so:
```suggestion
      <div className="buddies-smaller">{displayContext}</div>
```

Reply:
I'm just making sure they'll be displayed only if contextData exists, but I guess that's fine too.
Sometimes I'm getting errors that it may not exist, and I started doing that as a prevention.

Reply:
We want to avoid unneeded preventions, because later we would need to write down unneeded tests and it's making the code more complicated than it needs to be.